src/middleware/compression.rs

```diff
@@ -0,0 +1,204 @@
+use futures::future::FutureObj;
+
+use crate::body::Body;
+use crate::{middleware::RequestContext, Middleware, Response};
+use http::header::{HeaderValue, ACCEPT_ENCODING, CONTENT_ENCODING};
+
+use std::io::Read;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Encoding {
+    Gzip,
+    Deflate,
+    Identity,
+}
+
+impl Encoding {
+    fn parse(i: &str) -> Option<Encoding> {
+        use self::Encoding::*;
+        match i {
+            "gzip" => Some(Gzip),
+            "deflate" => Some(Deflate),
+            "identity" => Some(Identity),
+            _ => None,
```

Comment:
What is the difference between `Some(Identity)` and `None`? My understanding is that [they're both treated the same way](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding#Directives), so perhaps we could could combine these to make matching on them easier?

Reply:
Right now, the behavior for both is the same.
I suspect it might come in handy to have the explicit differentiation once quality value. 
A request containing `identity;q=0,unknown-compressio-algo` could (should?) respond [406 Not Acceptable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406)